mailserver/mailserver_db_leveldb.go

```diff
@@ -0,0 +1,169 @@
+package mailserver
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/status-im/status-go/params"
+	whisper "github.com/status-im/whisper/whisperv6"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/errors"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"time"
+)
+
+type LevelDBImpl struct {
```

Comment:
I'd say `type levelDB`: no need to state that that is an implementation and also no need to make it public, especially if the `db` interface isn't.